app/controllers/api/vcs.php

```diff
@@ -0,0 +1,1019 @@
+                continue;
+            }
+
+            $deployment = $dbForProject->createDocument('deployments', new Document([
+                '$id' => $deploymentId,
+                '$permissions' => [
+                    Permission::read(Role::any()),
+                    Permission::update(Role::any()),
+                    Permission::delete(Role::any()),
+                ],
+                'resourceId' => $functionId,
+                'resourceType' => 'functions',
+                'entrypoint' => $function->getAttribute('entrypoint'),
+                'commands' => $function->getAttribute('commands'),
+                'type' => 'vcs',
+                'vcsInstallationId' => $vcsInstallationId,
+                'vcsInstallationInternalId' => $vcsInstallationInternalId,
+                'vcsRepositoryId' => $repositoryId,
+                'vcsRepositoryDocId' => $vcsRepoId,
+                'vcsRepositoryDocInternalId' => $vcsRepoInternalId,
+                'vcsCommentId' => \strval($latestCommentId),
+                'vcsBranch' => $branchName,
+                'search' => implode(' ', [$deploymentId, $function->getAttribute('entrypoint')]),
+                'activate' => $activate,
+            ]));
+
+            $targetUrl = $request->getProtocol() . '://' . $request->getHostname() . "/console/project-$projectId/functions/function-$functionId";
+
+            if (!empty($SHA) && $function->getAttribute('vcsSilentMode', false) === false) {
+                $functionName = $function->getAttribute('name');
+                $projectName = $project->getAttribute('name');
+                $name = "{$functionName} ({$projectName})";
+                $message = 'Starting...';
+
+                $repositoryId = $resource->getAttribute('repositoryId');
+                $repositoryName = $github->getRepositoryName($repositoryId);
+                $owner = $github->getOwnerName($installationId);
+                $github->updateCommitStatus($repositoryName, $SHA, $owner, 'pending', $message, $targetUrl, $name);
+            }
+
+            $contribution = new Document([]);
+            if ($external) {
+                $pullRequestResponse = $github->getPullRequest($owner, $repositoryName, $pullRequest);
+
+                $contribution->setAttribute('ownerName', $pullRequestResponse['head']['repo']['owner']['login']);
+                $contribution->setAttribute('repositoryName', $pullRequestResponse['head']['repo']['name']);
+            }
+
+            $buildEvent = new Build();
+            $buildEvent
+                ->setType(BUILD_TYPE_DEPLOYMENT)
+                ->setResource($function)
+                ->setVcsContribution($contribution)
+                ->setDeployment($deployment)
+                ->setTargetUrl($targetUrl)
+                ->setSHA($SHA)
+                ->setProject($project)
+                ->trigger();
+
+            //TODO: Add event?
+        }
+    }
+};
+
+App::post('/v1/vcs/github/incomingwebhook')
```

Comment:
```suggestion
App::post('/v1/vcs/github/events')
```

Reply:
This would fit better in a REST structure, we can also change the description to `Create Event`